WIP: Distributed RADAR-Docker stack

yatharthranjan opened the pull request:
This breaks up the RADAR-base stack into different sets of components which can be deployed individually on separate hosts.

Some other changes from the single host RADAR-Docker:
- Updated to Confluent version 5.2.0
- Added Multi-host monitoring using Netdata.

**Please see the README for more info.**

TODO: 

- [ ] configure Portainer for using multiple docker remotes.
- [x] add a health check to commons for alerting.
- [ ] Abstract out the duplicate code shared in some components to /commons
- [x] Add authentication to netdata.
- [x] properly configure all the components to relay through the SMTP server provided.
- [x] Add monitoring for HDFS and Data components.
- [ ] Add README file to each component for docs on component level usage and configuration.
- [ ] Investigate adding a separate instance of Nginx webserver to frontend and admin components so can still access if proxies component is down.
- [ ] Investigate adding a load balancer config to Nginx for stateless services like gateway, rest-proxy and schema-registry.

yatharthranjan committed:
Initial Structure and HDFS implementation for distributed setup.

yatharthranjan committed:
Fix hdfs datanode install script

yatharthranjan committed:
Make script executable

yatharthranjan committed:
Add / in path

yatharthranjan committed:
Make script executable

yatharthranjan committed:
Fix util import

yatharthranjan committed:
Upgrade confluent to 5.2.0

yatharthranjan committed:
Make scripts executable

yatharthranjan committed:
Fix typo

yatharthranjan committed:
Add brokers to default network to expose ports

yatharthranjan committed:
add kafka connectors

yatharthranjan committed:
executable script

yatharthranjan committed:
Fix import

yatharthranjan committed:
copy template if absent

yatharthranjan committed:
fix path

yatharthranjan committed:
fix other paths

yatharthranjan committed:
import env

yatharthranjan committed:
build kafka-init before configuring connectors

yatharthranjan committed:
make hdfs url configurable

yatharthranjan committed:
Fix kafka advertised listeners

yatharthranjan committed:
import env

yatharthranjan committed:
Additional broker configs

yatharthranjan committed:
Fix kafka hosts

yatharthranjan committed:
Fix ports

yatharthranjan committed:
Revert changes to listeners

yatharthranjan committed:
Simplify configuration of brokers

yatharthranjan committed:
Fix variable checking

yatharthranjan committed:
Fix port

yatharthranjan committed:
ensure hdfs.url variable

yatharthranjan committed:
Add frontend services.

- management portal, smtp, rest sources authorizer

yatharthranjan committed:
fix docker-compose command

yatharthranjan committed:
executable keystore init

yatharthranjan committed:
Init kafka properly

yatharthranjan committed:
fix kafka-topics command

yatharthranjan committed:
More verbose output

yatharthranjan committed:
Fix command

yatharthranjan committed:
Fix _schemas topic creation

yatharthranjan committed:
add proxies - nginx, gateway, rest-proxy, schema-registry and radar-kafka-init

yatharthranjan committed:
remove unneeded config

yatharthranjan committed:
fix generate keys for gateway

yatharthranjan committed:
keep empty dir in git

yatharthranjan committed:
fix keystore path

yatharthranjan committed:
fix imports

yatharthranjan committed:
expose rest-proxy ports

yatharthranjan committed:
Add admin services

yatharthranjan committed:
Add dashboard services - rest-api, dashboard, hotstorage and backend-streams

yatharthranjan committed:
expose hotstorage ports

yatharthranjan committed:
Configurable mongodb host

yatharthranjan committed:
Fix variable subs

yatharthranjan committed:
Fix rest api path in nginx conf

yatharthranjan committed:
add data utilities

yatharthranjan committed:
fix script

yatharthranjan committed:
fix hdfs restructure scripts

yatharthranjan committed:
further fix script

yatharthranjan committed:
remove hadoop network

yatharthranjan committed:
fix syntax error

yatharthranjan committed:
more checks

yatharthranjan committed:
more verbose output

yatharthranjan committed:
print output directory

yatharthranjan committed:
fix script

yatharthranjan committed:
add hdfs restructure as a service

yatharthranjan committed:
fix indentation

yatharthranjan committed:
logging in connector

yatharthranjan committed:
add gitignore

yatharthranjan committed:
Local listener for inter broker communications

yatharthranjan committed:
Fix listeners

yatharthranjan committed:
expose ports

yatharthranjan committed:
remove duplicate config of brokers

yatharthranjan committed:
Plaintext for external listeners

yatharthranjan committed:
fix protocol map

yatharthranjan committed:
add redcap integration to frontend services

yatharthranjan committed:
minor fix

yatharthranjan committed:
keys init for rest api

yatharthranjan committed:
make script executable

yatharthranjan committed:
Confgurable radar-is.yml

yatharthranjan committed:
configurable radar-is for gateway

yatharthranjan committed:
expose redcap integration app

yatharthranjan committed:
bump version for rest-source-authorizer

yatharthranjan committed:
Fix urls

yatharthranjan committed:
Fix url path

yatharthranjan committed:
fix other url

yatharthranjan committed:
Initial setup for multi-host monitoring using Netdata

yatharthranjan committed:
fix variable check

yatharthranjan committed:
fix sed termination

yatharthranjan committed:
add file to sed

yatharthranjan committed:
run netdata as privileged

yatharthranjan committed:
add netdata proxy to webserver

yatharthranjan committed:
fix netdata proxy

yatharthranjan committed:
git add empty directory for netdata slave

yatharthranjan committed:
fix path

yatharthranjan committed:
misc minor fixes

yatharthranjan committed:
Make config volume mounts read-only

yatharthranjan committed:
fix volume mounts

yatharthranjan committed:
add PGID for container name resolution

yatharthranjan committed:
disable web console for netdata slaves

yatharthranjan committed:
Add netdata as slave to other components

yatharthranjan committed:
Remove unintended line.

yatharthranjan committed:
Add empty config folders

yatharthranjan committed:
Use netdata host instead or URL

- So can use same env variable for both nginx and netdata slave

yatharthranjan committed:
Add initial docs

yatharthranjan committed:
Add basic auth to netdata endpoint

yatharthranjan committed:
Add proxy for HDFS Namenode UI

yatharthranjan committed:
add sub_filter for hdfs namenode ui

yatharthranjan committed:
Add configurable hostname for datanode to distinguish between the nodes.

yatharthranjan committed:
Add check-health in commons to be used by all components.

yatharthranjan committed:
minor tweaks

yatharthranjan committed:
Fix docker-compose file path in health check

yatharthranjan committed:
minor fix

yatharthranjan committed:
use sensible defaults

yatharthranjan committed:
Only include --from if FROM_EMAIL is present

yatharthranjan committed:
make script executable

yatharthranjan committed:
add netdata slave to data component

yatharthranjan committed:
executable script

yatharthranjan committed:
minor updates

yatharthranjan committed:
remove sudo

yatharthranjan committed:
Initial support for netdata notifications

yatharthranjan committed:
fix copy template

yatharthranjan committed:
mount the conf to container

yatharthranjan committed:
Fix msmtp for netdata

yatharthranjan committed:
Add slack support for netdata alerts and documentation

yatharthranjan committed:
Add basic auth to hdfs namenode UI

yatharthranjan committed:
Fix portainer version

yatharthranjan committed:
add support for certificate renew and systemd timer

yatharthranjan committed:
postgres update script and other minor fixes

yatharthranjan committed:
Add documentation for admin component

yatharthranjan committed:
Adds further docs

yatharthranjan committed:
add missing config

yatharthranjan committed:
Add auth for rest source authoriser

yatharthranjan committed:
Fix streams networks

yatharthranjan committed:
Expose correct hdfs datanode ports and services

yatharthranjan committed:
Use latest Fitbit connector

yatharthranjan committed:
add role based recepients for netdata alerts

yatharthranjan committed:
Enable & configure nginx plugin for netdata

yatharthranjan committed:
Fix script

yatharthranjan committed:
Only register schemas if specified by the flag

yatharthranjan committed:
proper variable name

yatharthranjan committed:
add source statistics monitor

yatharthranjan committed:
Use host bind mounts for kafka logs

yatharthranjan committed:
Migrate Management portal to public key endpoint and comptible gateway version

yatharthranjan committed:
fix paths

yatharthranjan committed:
Add cors support for schema registry.

yatharthranjan committed:
add cors support to schema registry in no-ssl mode

yatharthranjan committed:
Merge remote-tracking branch 'origin/master' into distributed-dcompose

yatharthranjan committed:
BREAKING: Upgrade to latest Restructure 1.0.0

- This is a breaking change. Please check the migration guide of 
Restructure app from 0.5.x to 1.0.0.

yatharthranjan committed:
Minor updates to restructure

yatharthranjan committed:
fix keyword

yatharthranjan committed:
Add support for external postgres (eg- using AWS RDS)

yatharthranjan committed:
Make scripts executable

yatharthranjan committed:
Fix unnecessary directory check

yatharthranjan committed:
fix script path

yatharthranjan committed:
use connection URI for psql

yatharthranjan committed:
fix netdata config path

yatharthranjan committed:
move catalog server to proxies so can be exposed with nginx

yatharthranjan committed:
Fix catalog server location

yatharthranjan committed:
Remove rest sources authorizer from aws frontend services

yatharthranjan committed:
Update RADAR output to latest version

yatharthranjan committed:
Fix radar-output JAVA_OPTS

blootsvoets commented on distributed-dcompose-stack/data/docker-compose.yml:
```diff
@@ -18,7 +18,7 @@ services:
       - ${RESTRUCTURE_OUTPUT_DIR}:/output
       - ./etc/radar-output/restructure.yml:/etc/restructure.yml
     environment:
-      RADAR_HDFS_RESTRUCTURE_OPTS: ${RADAR_HDFS_RESTRUCTURE_OPTS}
```
No please do not use `JAVA_OPTS`, use `RADAR_OUTPUT_RESTRUCTURE_OPTS` instead.

yatharthranjan replied:
Ah ok, I see. Thanks

yatharthranjan committed:
Fix radar-output OPTS

yatharthranjan committed:
Update Redcap Integration app to latest
- adds support for attributes

yatharthranjan committed:
Update to latest Redcap integration on aws stack

yatharthranjan committed:
Minor fixes

yatharthranjan committed:
Latest redcap integration

yatharthranjan committed:
disable email alerts as using slack now

yatharthranjan committed:
update fitbit connector to latest version

yatharthranjan committed:
Use latest fitbit connector

yatharthranjan committed:
Add ksqlDB for processing of data.

yatharthranjan committed:
correct persistence format for ksqldb

yatharthranjan committed:
Add appserver alpha to services

yatharthranjan committed:
add missing netdata slave dir

yatharthranjan committed:
custom database name and rename generic

yatharthranjan committed:
latest appserver version

Hsankesara committed:
Added docker compose for mlflow tracking

Hsankesara committed:
updated dockerfile and docker-compose files

Hsankesara committed:
Added README and enviroment template files

Hsankesara committed:
Merge branch 'distributed-dcompose' of https://github.com/RADAR-base/RADAR-Docker into adding_mlflow_docker

yatharthranjan committed:
upgrade management portal and gatewa

Hsankesara committed:
Updated docker compose file

Hsankesara committed:
Merge pull request #239 from RADAR-base/adding_mlflow_docker

Added mlflow docker for tracking models

yatharthranjan committed:
upgrade to latest rest-source-auth version

yatharthranjan committed:
Merge branch 'distributed-dcompose' of https://github.com/RADAR-base/RADAR-Docker into distributed-dcompose

yatharthranjan committed:
remove basic auth for rest-sources-auth

yatharthranjan committed:
add feature db + jdbc connector

yatharthranjan committed:
Fix kafka host

yatharthranjan committed:
Fix volume mount path

yatharthranjan committed:
add push endpoint to the stack

yatharthranjan committed:
Expose push endpoint in the webserver

yatharthranjan committed:
Fix push endpoint service

yatharthranjan committed:
include .env vars

yatharthranjan committed:
Latest version of push endpoint

yatharthranjan committed:
minor fixes including size limit for garmin

yatharthranjan committed:
add support for ksql udfs/udafs

yatharthranjan committed:
add correct path for udfs

yatharthranjan committed:
add separate jdbc connector for aggregate streams

- This is needed as Avro field names are not case sensitive in KSQL. This matches the KSQL column name behavior.
- So field names are PROJECTID instead of projectId, and so on.

yatharthranjan committed:
minor updates

afolarin commented:
What is the status of this PR?

mpgxvii committed:
Merge branch 'distributed-dcompose' of https://github.com/RADAR-base/RADAR-Docker into distributed-dcompose

yatharthranjan commented:
Hi @afolarin, the components are stable and are working well and being used in production for quite some time. However, the documentation is incomplete (for individual components), thus this is not ready to be merged.

mpgxvii committed:
Add real time dashboard components to distributed stack

mpgxvii committed:
Merge branch 'dev' of https://github.com/RADAR-base/RADAR-Docker into sync-dev

mpgxvii committed:
Merge branch 'distributed-dcompose' of https://github.com/RADAR-base/RADAR-Docker into dashboard-distributed

mpgxvii committed:
Merge branch 'sync-dev' of https://github.com/RADAR-base/RADAR-Docker into dashboard-distributed

mpgxvii committed:
Add connector properties to gitignore

mpgxvii committed:
Fix env vars and install script

mpgxvii committed:
Merge pull request #251 from RADAR-base/sync-dev

Sync distributed stack branch with dev

mpgxvii committed:
Fix realtime dashboard docker-compose.yml

mpgxvii committed:
Update env template

mpgxvii committed:
Update README

yatharthranjan committed:
Merge pull request #249 from RADAR-base/dashboard-distributed

Add dashboard components to distributed stack

yatharthranjan committed:
add routing for grafana and mlflow

- Adds basic auth for mlflow endpoint

yatharthranjan committed:
Merge branch 'distributed-dcompose' of https://github.com/RADAR-base/RADAR-Docker into distributed-dcompose

yatharthranjan committed:
Add netdata monitoring to realtime dashboard

yatharthranjan committed:
Deprecate the old dashboard component

yatharthranjan committed:
Expose grafana for reverse proxy

yatharthranjan committed:
Fix basic auth for mlflow

yatharthranjan committed:
fix nginx.conf

yatharthranjan committed:
Remove deprecated API from nginx

yatharthranjan committed:
add smtp config to grafana

yatharthranjan committed:
grafana enable smtp

yatharthranjan committed:
add production appserver

yatharthranjan committed:
fix substitution

yatharthranjan committed:
use docker automated builds for kafka-init

yatharthranjan committed:
fix gateway healthcheck

yatharthranjan committed:
attempt to upgrade kafka-init

yatharthranjan committed:
add command for kafka-init

yatharthranjan committed:
update catalog server to latest

yatharthranjan committed:
add realtime inference consumer

yatharthranjan committed:
remove unused services
